### ISSUE TITLE
[ic-data-entity] Accessibility issue for data entity

### ISSUE THREAD

**Issue by GCHQ-Developer-530** [mi6/ic-ui-kit#1465]
## Summary of the bug 
When running accessibility tests in Cypress for data entity, there's an error to do with the an aria-parent not being set.

It can also be replicated running accessibility checks against our site.

It seems to be an issue with the `role="cell"` we've added to a `div` for the `end-component` slot. If that's removed then there's no errors, but the cell role must have been applied for a reason so worth looking into whether we need to replace it with a different role or if any of the screen reader functionality goes missing when it's removed.

## 📸 Screenshots or code 
<img width="1034" alt="Screenshot 2024-02-02 at 13 11 27" src="https://github.com/mi6/ic-ui-kit/assets/111882034/ce1e49fe-6e64-478f-940c-da3dfdee2f7e">

## 🧐 Expected behaviour 
It should pass Cypress accessibility tests.

**Comment by ASM995** [comment]
Please consider accessibility guidance if required.
